alt_e2eshark/e2e_testing/onnx_utils.py

```diff
@@ -29,6 +29,12 @@ def dtype_from_ort_node(node):
 
 def get_node_shape_from_dim_param_dict(node: onnxruntime.capi.onnxruntime_pybind11_state.NodeArg, dim_param_dict: Optional[dict[str, int]] = None):
     """Get the shape of a node, replacing any string dims with values from a dim_param_dict"""
+    if dim_param_dict and node.name in dim_param_dict.keys():
+        # Particularly used for models from ONNX Model Zoo,
+        # the dim_param_dict here should contain the shapes
+        # of the input nodes.
+        return dim_param_dict[node.name]
+
```

Comment:
The node name shouldn't be a key to the `dim_param_dict`. The `dim_param_dict` should take the string name of the dim parameter itself as a key. We could plumb through an alternative association of `input_node_name -> [static_shape, dtype]`, but this would be a separate dictionary.